cpp/src/Slice/Parser.h

```diff
@@ -395,19 +399,38 @@ class Builtin : public virtual Type
     virtual std::string typeId() const;
     virtual bool usesClasses() const;
     virtual size_t minWireSize() const;
+    virtual std::string getTagFormat() const;
     virtual bool isVariableLength() const;
 
+    bool isNumericType() const;
+    bool isIntegralType() const;
+
     Kind kind() const;
     std::string kindAsString() const;
+    static std::optional<Kind> kindFromString(std::string_view);
 
-    static const char* builtinTable[];
+    inline static const std::array<std::string, 12> builtinTable =
+    {
+        "byte",
+        "bool",
```

Comment:
Note that you would have to also swap the order of the `Kind` enumerators to match, since we depend on the ordering. But I think that the order should be swapped there anyways.

Reply:
I'd rather leave this swap to another PR as I would need to update the code generators accordingly.